slimta/relay/smtp/client.py

```diff
@@ -186,12 +186,18 @@ def _deliver(self, result, envelope):
         try:
             self._handle_encoding(envelope)
             self._send_envelope(rcpt_results, envelope)
-            self._send_message_data(envelope)
+            msg_result = self._send_message_data(envelope)
         except SmtpRelayError as e:
             result.set_exception(e)
             self._rset()
         else:
-            result.set(rcpt_results)
+            if rcpt_results is None:
```

Comment:
You are right. I will just drop this condition, it's going all the time in the `else` anyway, so it will be iso-functional with current PR.